exporters/otlp/otlpmetric/internal/oconf/options.go

```diff
@@ -89,16 +89,23 @@ func NewHTTPConfig(opts ...HTTPOption) Config {
 			URLPath:     DefaultMetricsPath,
 			Compression: NoCompression,
 			Timeout:     DefaultTimeout,
+			Insecure:    true,
```

Comment:
The default for this should be false according to the specification: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/protocol/exporter.md#configuration-options

Why is this set to `true` here?

Reply:
I just want to add a not that the specification is not worded perfectly.

> This option only applies to OTLP/gRPC when an endpoint is provided without the http or https scheme.

`insecure` is by default `true` when an endpoint is provided without the http or https scheme.

However, I think that @MrAlias is right that probably we should keep the defaults and not interpret the `Insecure` field if the HTTP scheme is defined.

Reply:
Thank you both. Looks like there is a bit of confusion from my end. I will re-read the specs with a rested eye over the weekend.

Reply:
@remychantenay 

[The wording for `insecure`](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.24.0/specification/protocol/exporter.md) says

> This option only applies to OTLP/gRPC when an endpoint is provided without the http or https scheme - OTLP/HTTP always uses the scheme provided for the endpoint.

My interpretation of this is that OTLP/HTTP endpoints *must* include a scheme (as noted n the specification).  But the scheme overrides whatever setting there may be for `Insecure` in the config (i.e. it's a noop setting for OTLP/HTTP).

For example, if the OTLP/HTTP endpoint starts with `http`, then the exporter should act as though `Insecure= true`, but if the scheme in the HTTP endpoint is `https`, it should act as if`Insecure=false`.  The wording seems to indicate that the user setting for `Insecure` shouldn't be changed, but rather the exporter should just use the appropriate setting regardless of the `Insecure` value in the config.

For OTLP/gRPC, there are more options for specifying the endpoint.  But *if* the gRPC endpoint includes an http or https scheme, then the same "ignore what the setting says and act as though it is whatever the scheme says" applies the same as in OTLP/HTTP.

But *if* the gRPC endpoint does *not* include an http or https scheme, then the `Insecure` setting should be taken into account and should default to `Insecure=false`.